test/system/executable/tests.py

```diff
@@ -103,10 +115,17 @@ def parse_options():
         action='store_true',
         help='Print test debugging info.'
     )
+    parser.add_argument(
+        '-s',
+        '--signal',
+        type=str,
```

Comment:
Instead checking if the input is string type, I'd use a function here that checks if the input is `sigint` or `sigterm` in lowercase and uppercase. 

```suggestion
        type=check_terminate_signal,
```